nomad/nomad_deployer.go

```diff
@@ -169,3 +174,35 @@ func compareJobs(old, new *nomadapi.Job) bool {
 
 	return false
 }
+
+// addTLSInfo will inject Nomad TLS certificate info to updater job if the TLS certificate is provided
+func addTLSInfo(job *nomadapi.Job) {
+	// Always make sure that job.ID matches with the job ID in nomad mustache template on server side
+	// Suggest to use a job name with format "portainer-updater-{uuid}""
+	targetJobName := "portainer-updater"
+	if *job.ID == targetJobName && *job.TaskGroups[0].Name == targetJobName && job.TaskGroups[0].Tasks[0].Name == targetJobName {
```

Comment:
do we need to use `*job` or `job`? because they have both in this line

Reply:
also, might be good to check that there are TaskGroups and Tasks (i.e `len()>0`)

Reply:
> do we need to use `*job` or `job`? because they have both in this line

Yes. It is because the type of `job.ID` and `job.TaskGroups[0].Name` is `*string` and the type of `job.TaskGroups[0].Tasks[0].Name` is `string`

> also, might be good to check that there are TaskGroups and Tasks (i.e `len()>0`)

Good point